MapData.py

```diff
@@ -0,0 +1,41 @@
+import os
+import threading
+from enum import Enum
+
+import matplotlib.pyplot as plt
+from scipy.misc import imresize
+
+from detail import LOCAL
+
+class MapDataFieldNamesEnum(Enum):
+    ZOOM = 'zoom'
+    RADIUS = 'radius'
+    IMAGE = 'image'
+    MARKER_PATH = 'marker path'
+    MARKER = 'marker'
+    ANNO_BOX = 'annotation box'
+    LOCATION = 'location'
+
+class MapData:
+    def __init__(self):
+        self.lock = threading.Lock()
+
+        # Map UI attributes
+        self.mapData = {}
+        self.mapData[MapDataFieldNamesEnum.ZOOM.value] = 20
+        self.mapData[MapDataFieldNamesEnum.RADIUS.value] = 0.1
+        self.mapData[MapDataFieldNamesEnum.IMAGE.value] = None
+        self.mapData[MapDataFieldNamesEnum.MARKER_PATH.value] = os.path.join(LOCAL, "marker.png")
+        # TODO: imresize removed in latest scipy since it's a duplicate from "Pillow". Update and replace.
+        self.mapData[MapDataFieldNamesEnum.MARKER.value] = imresize(plt.imread(self.mapData[MapDataFieldNamesEnum.MARKER_PATH.value]), (12, 12))
+        self.mapData[MapDataFieldNamesEnum.ANNO_BOX.value] = None
+        self.mapData[MapDataFieldNamesEnum.LOCATION.value] = None
```

Comment:
At first glance, this looks fine, just another map. But after some consideration, it seems like this map is really just trying to be a POD class/struct. So I think to reduce the overhead of looking up in a map and to keep the code simpler, we should just turn it into a POD class.

But then the question is, do we really need ANOTHER class? MapData was already the map data class, so it would seem like we could just make all of these values fields of MapData. However I'm guessing that maybe the original intent with using the map in the first place was to do a sort of "time capture" so that when we plot the map we ensure that all the info came from the same time? But thats kinda what mutexes are for so I think I have a solution for this. Lets arrange to discus this more.

Reply:
After discussion today it seems like we have still retained the class with a lot of changes but we can still talk about this further.